### ISSUE TITLE
The problem about data partitioning

### ISSUE THREAD

**Issue by jiashizuo** [Graph-and-Geometric-Learning/HyBRiD#4]
Thank you very much for the open source code!

Does the code have a partitioned validation set？I only saw  “train_dataloader” and “test_dataloader” in dataset.py.
![image](https://github.com/user-attachments/assets/5e49e551-81e0-4265-9fea-160b807010ed)
This part of the operation involves merging the validation set into the training set. "val_ids" is not used.

Was all the data （Including training set）used during testing?
![image](https://github.com/user-attachments/assets/dc558af8-0f25-4eef-bb27-4d0f2b526035)

**Comment by Boltzmachine** [comment]
Hi Thanks for your interest in our work!

https://github.com/Graph-and-Geometric-Learning/HyBRiD/blob/cb37d40bd2b786035814bee6880fb086a59f88d3/src/dataset.py#L179-L181
For your first question, as stated in the comments, we search the hyperparameters under the train/val/test split and then train the model with the best hyperparameters on both train and val sets considering the limited dataset size (and actually we didn't do much hyperparameter search). You can implement your own validation strategy. We recommend cross-validation given the limited dataset size. But it is not necessary if you want to run the experiments since we do not use it during training. I will also add this procedure in the future.

For your second question,
The test data loader and the test step are misleading. They are actually not `test` in our experiment settings. Instead, it is more like an inference step. We use CPM to perform the actual evaluation.
https://github.com/Graph-and-Geometric-Learning/HyBRiD/blob/cb37d40bd2b786035814bee6880fb086a59f88d3/src/model.py#L175

I will try to make the code more clear. Thanks for you feedback!
